azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusUtils.java

```diff
@@ -0,0 +1,81 @@
+package azkaban.executor.container.watch;
+
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_APP_FAILURE;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_COMPLETED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_DELETED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_INIT_FAILURE;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_READY;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_REQUESTED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_SCHEDULED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_UNEXPECTED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_UNSET;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Contains routines for validation of {@link AzPodStatus} transitions and any other processing
+ * required for the statuses.
+ */
+public class AzPodStatusUtils {
+  private static final Set<AzPodStatus> finalStatuses = EnumSet.of(
+      AZ_POD_COMPLETED,
+      AZ_POD_INIT_FAILURE,
+      AZ_POD_APP_FAILURE,
+      AZ_POD_DELETED,
+      AZ_POD_UNEXPECTED
+  );
+
+  // A key in this map is the 'target' status and the value is the set of 'source statuses' from
+  // which transition the the 'target' (key) status is not permitted and and event sequence
+  // corresponding to these transitions should never be received.
+  private static final Map<AzPodStatus, Set<AzPodStatus>> invalidSourceStatusMap =
+      buildInvalidStatusMap();
+
+  // Build the invalid status static map.
+  private static Map<AzPodStatus, Set<AzPodStatus>> buildInvalidStatusMap() {
+    ImmutableMap.Builder builder = ImmutableMap.builder()
+        .put(AZ_POD_READY, finalStatuses);
+
+    Set<AzPodStatus> mutableInvalidStatuses = new HashSet<>(finalStatuses);
```

Comment:
Can you please move builder for each status in separate method or something? 
getInvalidStatusesForAZPodReady and it will return set of invalid statuses for it. It will be useful to maintain this map as well as in future.